datafusion/optimizer/src/optimize_projections/mod.rs

```diff
@@ -682,7 +682,8 @@ fn split_join_requirements(
         | JoinType::Left
         | JoinType::Right
         | JoinType::Full
-        | JoinType::LeftMark => {
+        | JoinType::LeftMark 
+        | JoinType::RightMark => {
             // Decrease right side indices by `left_len` so that they point to valid
             // positions within the right child:
             indices.split_off(left_len)
```

Comment:
Is this really correct? Feels like we will assign the wrong indices to the left here for `RightMark`? Should `RightMark` be moved to `RightSemi` and `RightAnti`. Maybe we should for `LeftMark` as well for clarity (and maybe that also fixes some edge case?)